dedupe/predicate_functions.py

```diff
@@ -0,0 +1,268 @@
+import re
+from itertools import chain
+from math import copysign, floor, log10
+from typing import Any, Sequence, Set, Tuple, Union
+
+from doublemetaphone import doublemetaphone
+
+from dedupe.cpredicates import initials, ngrams, unique_ngrams  # noqa: F401
+
+# Each predicate function returns a collection of strings. Since order
```

Comment:
this will need to be revised.

Reply:
I completely removed this comment as now we always return frozensets.